src/Framework/MSBuildEventSource.cs

```diff
@@ -398,6 +398,18 @@ public void ExecuteTaskReacquireStop(string taskName, int taskID)
         {
             WriteEvent(52, taskName, taskID);
         }
+
+        [Event(53)]
+        public void ProjectGraphConstructionStart()
```

Comment:
Decided to not send any data in. The only relevant data that I could send in would be a super long string concatenating either the graph entry point project files, or the graph entry point project files together with their global properties, but this can be inferred from the evaluation events between the graph construction start / end.

Reply:
Just what I was about to ask about 👍

Reply:
Although you should be able to send a variable number of strings rather than concatenating everything, so it should be performant anyway.

Reply:
The project path strings are wrapped in other objects, so it requires constructing a collection to put them in (not really that perf impacting, really, only happens once). Global properties for each entry project would be more expensive to format into strings, but again, probably not that impactful since it runs once. On the other hand, all this information is available from the evaluation events that happen between graph construction start/stop, so I'm unsure whether it's worth putting on. Let me know which option you prefer (project paths, project paths + global properties for each one, or nothing) and I'll add it in

Reply:
If you wrap it in `MSBuildEventSource.Log.IsEnabled()` calls, you'll only do the calculations when someone is actually interested in the data, so it would have ~no effect on perf. That also means that my tendency would be to log everything everywhere—no point in hiding information. I don't feel that strongly, though, personally.

Reply:
Cool, pushed a commit with the changes.